lib/context.js

```diff
@@ -266,6 +266,53 @@ api.process = async ({
       defined.set('@propagate', true);
     }
 
+    // handle @import
+    if('@import' in ctx) {
+      const value = ctx['@import'];
+      if(activeCtx.processingMode === 'json-ld-1.0') {
+        throw new JsonLdError(
+          'Invalid JSON-LD syntax; @import not compatible with ' +
+          activeCtx.processingMode,
+          'jsonld.SyntaxError',
+          {code: 'invalid context member', context: ctx});
+      }
+      if(!_isString(value)) {
+        throw new JsonLdError(
+          'Invalid JSON-LD syntax; @import must be a string.',
+          'jsonld.SyntaxError',
+          {code: 'invalid @import value', context: localCtx});
+      }
+
+      // resolve contexts
+      const resolvedImport = await options.contextResolver.resolve({
+        context: value,
+        documentLoader: options.documentLoader,
+        base: options.base
+      });
+      if(resolvedImport.length !== 1) {
+        throw new JsonLdError(
+          'Invalid JSON-LD syntax; @import must reference a single context.',
+          'jsonld.SyntaxError',
+          {code: 'invalid remote context', context: localCtx});
+      }
+      const importCtx = resolvedImport[0].document;
+      if('@import' in importCtx) {
+        throw new JsonLdError(
+          'Invalid JSON-LD syntax; imported context must not include @import.',
+          'jsonld.SyntaxError',
+          {code: 'invalid context member', context: localCtx});
+      }
+
+      // merge ctx into importCtx and replace rval with the result
+      for(const key in importCtx) {
+        if(!(key in ctx)) {
```

Comment:
Avoid problem if importCtx has default Object property term names like 'toString':
```suggestion
        if(!ctx.hasOwnProperty(key)) {
```